src/test/java/com/cloudant/tests/HttpProxyTest.java

```diff
@@ -63,12 +63,10 @@ public void proxyConfiguration() throws Exception {
         for (String line : server.getLastInputRequestLines()) {
             if (line.contains("Proxy-Authorization")) {
                 foundProxyAuthHeader = true;
-                Matcher m = Pattern.compile("Proxy-Authorization\\s*:\\s*(.*)", Pattern
+                Matcher m = Pattern.compile("Proxy-Authorization: Basic (.*)", Pattern
                         .CASE_INSENSITIVE).matcher(line);
                 assertTrue("The Proxy-Authorization header should match the pattern", m.matches());
-                assertEquals("There should be 2 groups, one for the header key and one for the " +
-                        "value", 2, m.groupCount());
-
+                assertEquals("There should be 1 group for the value", 1, m.groupCount());
```

Comment:
The test did not fail before because the `Proxy-Authentication` header wasn't being added due to the bug in `BasicAuthInterceptor` that didn't use the header key from the subclass. This made the test bypass all the matching code and the lack of an assertion on the boolean `foundProxyAuthHeader` meant it didn't fail even though the header was missing. These changes mean the test would now fail for the lack of header or broken regex.
The test uses the `SimpleHttpServer` as a proxy to verify that the requests are routed to the proxy and contain the expected auth header, but it doesn't actually do any real proxying of requests.